src/control/lib/netdetect/netdetect_integration_test.go

```diff
@@ -34,53 +33,73 @@ import (
 	"testing"
 	"time"
 
+	"github.com/pkg/errors"
+
 	"github.com/daos-stack/daos/src/control/common"
 	"github.com/daos-stack/daos/src/control/logging"
 )
 
 const maxConcurrent = 1000
 
-func getSocketID(t *testing.T, pid int32, wg *sync.WaitGroup) {
+func getSocketID(t *testing.T, pid int32, wg *sync.WaitGroup, ctx context.Context) error {
 	defer wg.Done()
-
-	netCtx, err := Init(context.Background())
-	defer CleanUp(netCtx)
-	common.AssertEqual(t, err, nil, fmt.Sprintf("Failed to initialize NetDetectContext: %v", err))
-
-	numaNode, err := GetNUMASocketIDForPid(netCtx, int32(pid))
-	common.AssertEqual(t, err, nil, fmt.Sprintf("GetNUMASocketIDForPid error on NUMA %d / pid %d", numaNode, pid))
+	numaNode, err := GetNUMASocketIDForPid(ctx, int32(pid))
+	if err != nil {
+		return errors.Errorf("GetNUMASocketIDForPid error on NUMA %d / pid %d / error: %v", numaNode, pid, err)
+	}
+	return nil
 }
 
-// TestConcurrentGetNUMASocket launches maxConcurrent go routines
-// that concurrently access the hwloc topology.  This test verifies that there are
-// no race conditions when the topology is initialized on each access
+// TestConcurrentGetNUMASocket launches go routines
+// that concurrently call GetNUMASocketIDForPid() by itself to verify that there are
+// no race conditions
 func TestConcurrentGetNUMASocket(t *testing.T) {
 	_, buf := logging.NewTestLogger(t.Name())
 	defer common.ShowBufferOnFailure(t, buf)
 
+	netCtx, err := Init(context.Background())
+	defer CleanUp(netCtx)
+	common.AssertEqual(t, err, nil, fmt.Sprintf("Failed to initialize NetDetectContext: %v", err))
+
 	var wg sync.WaitGroup
 	pid := syscall.Getpid()
 	rand.Seed(time.Now().UnixNano())
-	for i := 0; i < maxConcurrent; i++ {
+	r := rand.Intn(1500)
+	for i := 0; i < maxConcurrent+r; i++ {
 		wg.Add(1)
-		go func(n int32) {
+		go func(n int32, ctx context.Context) {
```

Comment:
Here, as well. I know this is test code, but it should follow convention.

Reply:
fixed